account_invoice_pricelist_sale_stock/models/stock_picking.py

```diff
@@ -13,8 +13,9 @@ def _get_invoice_vals(self, key, inv_type, journal_id, move):
         inv_vals = super(StockPicking, self)._get_invoice_vals(
             key, inv_type, journal_id, move
         )
-        if self.sale_id and self.sale_id.pricelist_id:
+        sale = move.picking_id.sale_id
+        if sale and sale.pricelist_id:
```

Comment:
You can put directly `if sale.pricelist_id:`

Reply:
no because sale is not defined always.

Reply:
Although not set, new API returns an empty recordset and then you can query the same `sale.pricelist_id`

Reply:
Ow, thanks for the Tips !
I did'nt know.

regards.